### ISSUE TITLE
Consolidate all amazonka-* libraries into a single amazonka library

### ISSUE THREAD

**Issue by brendanhay** [brendanhay/amazonka#671]
This is a tracking issue for Cabal's (and Hackage's)  "Multiple public libraries" feature.

See:

* Announcement: https://fgaz.me/posts/2019-11-14-cabal-multiple-libraries/
* Original PR: https://github.com/haskell/cabal/pull/5526
* Cabal tracking issue: https://github.com/haskell/cabal/issues/5660
* Hackage support (supposed): https://github.com/haskell/hackage-server/issues/852

There are currently 140~ services configured and roughly that number still unconfigured (ie. not generated) that I've held off generating in the hope the feature would eventually be available/usable. Since it doesn't appear like it's there yet, I'd like to have a more concrete estimate of when (if ever) the feature will be usable on Hackage, otherwise generating the services into separate packages should resume.

Please link and discuss any additional updates and relevant issues.

**Comment by ketzacoatl** [comment]
@endgame do you mean https://github.com/commercialhaskell/stack/issues/5256?

**Comment by endgame** [comment]
Yes, thanks.

**Comment by brendanhay** [comment]
[This blog post](https://www.snoyman.com/blog/babies-oss-maintenance/#special-call-out-stack) may (or may not) be an indication of how likely such features will be supported by stack.

**Comment by endgame** [comment]
Yeah. That's pretty recent so can probably be taken as true. If we move to multiple public libraries in a single package (if/when Hackage supports that), we may create more interest in stack gaining that support, or we may end up locking amazonka out of the stack universe for a long time.

Also note: the nix world is downstream from the stackage world - most of their stuff is built on stackage snapshots. If you want to do this, we should probably liase with them first. I have no idea if `cabal2nix` etc support multiple public libraries, and we may inflict a great deal of pain.

**Comment by istathar** [comment]
I happen to know Stack is getting some additional maintainers shortly, which is great news. Michael came across a tad pessimistically in that post, but do I gather things like the multiple public libraries idea have caused a lot of pain.

**Comment by endgame** [comment]
That's good news. Still need to check where hackage is at with all this stuff, as well as cabal2nix and haskell.nix. No point getting a whizz-bang new version of Amazonka out the door if nobody can use it =(

**Comment by michaelpj** [comment]
> haskell.nix

Just to say: haskell.nix works fine with multiple public libraries.
